tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SDK/RuntimePipelineInvokeAsyncIntegration.cs

```diff
@@ -73,19 +73,22 @@ internal static TResponse OnAsyncMethodEnd<TTarget, TResponse>(TTarget instance,
             {
                 if (state.State is IExecutionContext { RequestContext.Request: { } request })
                 {
-                    var uri = request.Endpoint;
-                    var absolutePath = uri?.AbsolutePath;
-                    var path = request.ResourcePath switch
+                    tags.HttpMethod = request.HttpMethod?.ToUpperInvariant();
+                    if (tags.HttpUrl is null)
                     {
-                        null => absolutePath,
-                        string resourcePath when absolutePath == "/" => resourcePath,
-                        string resourcePath => UriHelpers.Combine(absolutePath, resourcePath),
-                    };
+                        var uri = request.Endpoint;
+                        var absolutePath = uri?.AbsolutePath;
+                        var path = request.ResourcePath switch
+                        {
+                            null => absolutePath,
+                            string resourcePath when absolutePath == "/" => resourcePath,
+                            string resourcePath => UriHelpers.Combine(absolutePath, resourcePath),
+                        };
 
-                    // The request object is populated later by the Marshaller,
-                    // so we wait until the method end callback to read it
-                    tags.HttpMethod = request.HttpMethod?.ToUpperInvariant();
-                    tags.HttpUrl = $"{uri?.Scheme}{Uri.SchemeDelimiter}{uri?.Authority}{path}";
```

Comment:
The only actual changes here are:
- move setting `HttpMethod` to the top
- Don't do the work to calculate `HttpUrl` if it's already been set - defer to the existing value instead (set in `HttpWebRequestMessage`)